components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1007,6 +1015,215 @@ public void updateResidentIdP(IdentityProvider identityProvider, String tenantDo
 
     }
 
+    /**
+     * Get all basic identity provider information.
+     *
+     * @param limit        limit per page.
+     * @param offset       offset value.
+     * @param filter       filter value for IdP search.
+     * @param sortOrder    order of IdP ASC/DESC.
+     * @param sortBy       the column value need to sort.
+     * @param tenantDomain tenant domain whose IdP names are requested.
+     * @return Identity Provider's Basic Information array {@link IdpSearchResult}.
+     * @throws IdentityProviderManagementException Error when getting list of Identity Providers.
+     */
+    @Override
+    public IdpSearchResult getIdPs(int limit, int offset, String filter, String sortOrder, String sortBy,
```

Comment:
change limit and offset to 'Integer' type and validate whether they are null before processing